diff --git a/node_modules/@dylankenneally/react-native-ssh-sftp/ios/RNSSHClient.m b/node_modules/@dylankenneally/react-native-ssh-sftp/ios/RNSSHClient.m
index 069031e..eddf58b 100644
--- a/node_modules/@dylankenneally/react-native-ssh-sftp/ios/RNSSHClient.m
+++ b/node_modules/@dylankenneally/react-native-ssh-sftp/ios/RNSSHClient.m
@@ -68,7 +68,7 @@ - (BOOL)isSFTPConnected:(NMSFTP *)sftpSesion
             [session authenticateByInMemoryPublicKey:[passwordOrKey objectForKey:@"publicKey"] privateKey:[passwordOrKey objectForKey:@"privateKey"] andPassword:[passwordOrKey objectForKey:@"passphrase"]];
 
             if (session.isAuthorized) {
-                SSHClient* client = [[SSHClient alloc] init];
+                SSHClient* client = [[SSHClient alloc] initWithHost:username withUsername:host];
                 client._session = session;
                 client._key = key;
                 [[self clientPool] setObject:client forKey:key];
diff --git a/node_modules/@dylankenneally/react-native-ssh-sftp/ios/SSHClient.h b/node_modules/@dylankenneally/react-native-ssh-sftp/ios/SSHClient.h
index 3948530..dda0684 100644
--- a/node_modules/@dylankenneally/react-native-ssh-sftp/ios/SSHClient.h
+++ b/node_modules/@dylankenneally/react-native-ssh-sftp/ios/SSHClient.h
@@ -18,4 +18,5 @@
 - (void) startShell:(NSString *)ptyType error:(NSError **)error;
 - (void) sftpDownload:(NSString *)path toPath:(NSString *)filePath error:(NSError **)error;
 - (BOOL) sftpUpload:(NSString *)filePath toPath:(NSString *)path;
+- (instancetype)initWithHost:(NSString *)host withUsername:(NSString *)username;
 @end
diff --git a/node_modules/@dylankenneally/react-native-ssh-sftp/ios/SSHClient.m b/node_modules/@dylankenneally/react-native-ssh-sftp/ios/SSHClient.m
index f6ab8f6..b766f9e 100644
--- a/node_modules/@dylankenneally/react-native-ssh-sftp/ios/SSHClient.m
+++ b/node_modules/@dylankenneally/react-native-ssh-sftp/ios/SSHClient.m
@@ -13,9 +13,9 @@ @implementation SSHClient
 int downloadedPerc = 0;
 int uploadedPerc = 0;
 
-- (instancetype)init {
+- (instancetype)initWithHost:(NSString *)host withUsername:(NSString *)username {
     if ((self = [super init])) {
-        _session = [[NMSSHSession alloc] init];
+        _session = [[NMSSHSession alloc] initWithHost:host andUsername:username];
         _sftpSession = [[NMSFTP alloc] init];
         _key = [[NSString alloc] init];
         _downloadContinue = false;
